Vita snake

slobters opened the pull request:
# New Port for { vitaSnake }

## Game Information
- **Title**: { vitaSnake }
- **URL**: { https://github.com/Grzybojad/vitaSnake }

## Submission Requirements

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [ X] AmberELEC
- [X ] ArkOS
- [ X] ROCKNIX
- [ X] muOS

### Resolution Tests
Test all major resolutions:
- [X ] 480x320
- [X ] 640x480
- [X ] 720x720 (RGB30)
- [X ] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

cdeletre commented on ports/vitasnake/cover.png:
it's possible to optimize the png size with pngquant (down to 56 kB) or online with https://squoosh.app/

cdeletre commented on ports/vitasnake/log.txt:
This file can be removed

cdeletre commented on ports/vitasnake/screenshot.png:
image weight can be reduced to 92 kB with pngquant. You can also use https://squoosh.app/

cdeletre commented on ports/vitasnake/splash.png:
It's a duplicate of `gamedata/splash.png` so maybe it can be removed ?

cdeletre commented on ports/vitasnake/vitaSnake.sh:
```diff
@@ -0,0 +1,52 @@
+export GMLOADER_PLATFORM="os_linux"
+
+# We log the execution of the script into log.txt
+exec > >(tee "$GAMEDIR/log.txt") 2>&1
```
Old logging method we now use a newer version which overwrites itself

`"$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1`

cdeletre commented on ports/vitasnake/vitasnake.md:
This file should be renamed as `README.md`

cdeletre commented:
The port file/folder structure must be revised, it should be something like this:
```
spookynakki % tree
.
├── README.md
├── Vitasnake.sh
├── cover.png
├── gameinfo.xml
├── port.json
├── screenshot.png
└── vitasnake
    ├── game.apk
    ├── gamedata
    ├── gmloader
    └── libs
```

cdeletre reviewed (changes requested):
The port file/folder structure must be revised, it should be something like this:
```
Portmaster-New
└──ports
    └──vitasnake
        ├── README.md
        ├── Vitasnake.sh
        ├── cover.png
        ├── gameinfo.xml
        ├── port.json
        ├── screenshot.png
        └── vitasnake
            ├── game.apk
            ├── gamedata
            ├── gmloader
            ├── ...
            └── libs
```

Also `port.json` is missing

Cebion commented on ports/vitasnake/port.json:
```diff
@@ -0,0 +1,33 @@
+    "exp": false,
+    "runtime": null,
+    "reqs": [
+      "power"
```
This requirement is set if it only works on cpus above RK3326 is this correct?

slobters commented:
I wasn't sure how to test this so I set it like that.
But this game should be able to run on something like RK3326.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

On Thu, Sep 19, 2024 at 11:30 PM Cebion ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In ports/vitasnake/port.json
> <https://github.com/PortsMaster/PortMaster-New/pull/667#discussion_r1767619633>
> :
>
> > +      "Slobters"
> +    ],
> +    "desc": "A snake game with smooth rotation made for the PS Vita with LibVita2D. Turn left and right to collect as many apples as you can, but avoid biting your tail, or you will lose the game.",
> +    "desc_md": null,
> +    "inst": "Ready to run !",
> +    "inst_md": null,
> +    "genres": [
> +      "action",
> +      "arcade"
> +    ],
> +    "image": null,
> +    "rtr": true,
> +    "exp": false,
> +    "runtime": null,
> +    "reqs": [
> +      "power"
>
> This requirement is set if it only works on cpus above RK3326 is this
> correct?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PortsMaster/PortMaster-New/pull/667#pullrequestreview-2316783001>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7WIAQ3OUCEC6JOST6MXXSLZXM7BBAVCNFSM6AAAAABOKH6L3GVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMJWG44DGMBQGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

JanTrueno commented on ports/vitasnake/cover.png:
Cover needs to be 640x480.

slobters replied:
Cover is 640x480 from 16:9 image with black bars.
Shoul I stretch or croop original image to 640x480?

slobters replied:
Oh wait cover, it's always 400x300 in the ports when I download them.
Screenshot is fine and cover need to be 640x480?
Why covers have 400x300 inside ports then?
Covers get shrinked for muOS?

JanTrueno replied:
Not sure which ports you are talking about, but both screenshot and cover have to be 640x480 :)

JanTrueno replied:
Maybe they do get shrunk for MuOS, thats very possible. I dont know though.

slobters replied:
I will make new cover later today.
Thanks for help.

Cebion reviewed (changes requested):
The file structure is wrong.
In the top vitasnake folder you only need readme, cover, gameinfo.xml, portj.json, screenshot and the .sh

slobters commented:
Fixed!
I moved the licence file into the second vitasnake folder. Is that ok?

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

On Mon, Sep 23, 2024 at 9:17 AM Cebion ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> The file structure is wrong.
> In the top vitasnake folder you only need readme, cover, gameinfo.xml,
> portj.json, screenshot and the .sh
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PortsMaster/PortMaster-New/pull/667#pullrequestreview-2321298083>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7WIAQ4U3AZ3JLVCPXR2JOTZX657XAVCNFSM6AAAAABOKH6L3GVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMRRGI4TQMBYGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
